libs/ui/src/utils/dapp.utils.ts

```diff
@@ -27,18 +28,31 @@ const createErrorMessage = (id: string) => ({
   target: 'oko-inpage'
 });
 
-export const sendResponseToDAppAndClosePopup = <T>(messageID: string, result: T) => {
-  tabs.query({ active: true }).then(queryTabs => {
-    if (queryTabs[0].id !== undefined) {
-      tabs.sendMessage(queryTabs[0].id, createDAppResponse(messageID, result));
+export const sendResponseToDAppAndClosePopup = <T>(messageID: string, result: T, origin?: string) => {
+  tabs.query({}).then(queryTabs => {
+    if (isDefined(origin)) {
+      const dAppTab = queryTabs.find(tab => tab.url?.includes(origin));
+      if (isDefined(dAppTab) && isDefined(dAppTab.id)) {
+        tabs.sendMessage(dAppTab.id, createDAppResponse(messageID, result));
+      }
+    } else {
+      if (queryTabs[0].id !== undefined) {
+        tabs.sendMessage(queryTabs[0].id, createDAppResponse(messageID, result));
+      }
```

Comment:
why `origin` is optional parameter?

Reply:
reuse same code from `sendResponseToDAppAndClosePopup`, `sendErrorToDAppAndClosePopup` and `sendNotificationToDApp`